test/validation/api/ipsec/ipsec.c

```diff
@@ -998,6 +1153,39 @@ int ipsec_check_out(const ipsec_test_part *part, odp_ipsec_sa_t sa,
 		} else {
 			/* IPsec packet */
 			CU_ASSERT_EQUAL(0, odp_ipsec_result(&result, pkto[i]));
+
+			if (part->out[i].sa_expiry != IPSEC_TEST_EXPIRY_IGNORED &&
+			    !sa_expiry_notified) {
```

Comment:
I think in other cases than inline-out-with-soft-expiry this should always check the warning/error bits as all packets should have them set. Now I think the validation test would pass if hard expiry gets reported for some packets but not all packets after the SA expiry.

Reply:
I didn't quite understand the first part of the question. Did you mean to keep a CU_ASSERT for warn bits similar to the error bits like:
                        CU_ASSERT_EQUAL(part->out[i].status.warn.all,
                                        result.status.warn.all)

> Now I think the validation test would pass if hard expiry gets reported for some packets but not all packets after the SA expiry.
In the "else if {}"part, we are having CU_ASSERT on hard_exp_[bytes|packets] for every subsequent packet once sa_expiry_notified is set for the first time in the "if {}" part of the if - else if.

As mentioned earlier, this part of the code is now moved to ipsec_check_sa_expiry() function for better code indentation.